File: lib_common/src/main/java/com/example/common/widget/dialog/LoadingDialog.java
package com.example.common.widget.dialog;

import android.content.Context;

import androidx.annotation.NonNull;

import com.example.common.R;
import com.example.common.base.BaseDialog;
import com.example.common.databinding.ViewDialogLoadingBinding;

/**
 * Created by wyb on 2017/6/28.
 * 加载动画view
 */
public class LoadingDialog extends BaseDialog<ViewDialogLoadingBinding> {

    public LoadingDialog(@NonNull Context context) {
        super(context, R.style.loadingStyle);
        initialize();
    }

    @Override
    protected int getLayoutResID() {
        return R.layout.view_dialog_loading;
    }

    public void show(boolean flag) {
        setCancelable(flag);
        if (!isShowing()) {
            show();
        }
    }

    public void hide() {
        if (isShowing()) {
            dismiss();
        }
    }

}

File: lib_common/src/main/java/com/example/common/widget/dialog/LoadingDialog.kt
package com.example.common.widget.dialog

import androidx.annotation.ColorInt
import androidx.fragment.app.FragmentActivity
import com.example.common.R
import com.example.common.base.BaseDialog
import com.example.common.databinding.ViewDialogLoadingBinding
import com.example.common.utils.function.color

/**
 * Created by wyb on 2017/6/28.
 * 加载动画view
 */
class LoadingDialog(activity: FragmentActivity) : BaseDialog<ViewDialogLoadingBinding>(activity, R.style.LoadingStyle, 90, 90, hasAnimation = false) {

    init {
        setColorFilter(color(R.color.bgWhite))
    }

    override fun show() {
        super.show()
        mBinding?.pbProcess?.isIndeterminate = true
    }

    override fun dismiss() {
        super.dismiss()
        mBinding?.pbProcess?.isIndeterminate = false
    }

    fun setColorFilter(@ColorInt color: Int) {
        mBinding?.pbProcess?.indeterminateDrawable = mBinding?.pbProcess?.indeterminateDrawable?.mutate()?.apply {
            setTint(color)
        }
    }

}